internal/worker/server.go

```diff
@@ -46,13 +50,26 @@ type JobStatus struct {
 	Canceled bool
 }
 
+type IdentityHeader struct {
```

Comment:
I suggest making this struct private. Actually, it could live inside the `VerifyIdentityHeader` method but both is fine with me.

Reply:
We might wanna use it in handlers at some point. I'm not sure. But until then it can live inside the method